### PR TITLE
Store Apollon Diagrams per course

### PR DESCRIPTION
**Pull request by sjagla27**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I added multiple integration tests (Spring) related to the features
- [x] Server: I added `@PreAuthorize` and check the course groups for all new REST Calls (security)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Server: I documented the Java code using JavaDoc style.
- [x] Client: I added `authorities` to all new routes and check the course groups for displaying navigation elements (links, buttons)
- [x] Client: I documented the TypeScript code using JSDoc style.
- [x] Client: I added multiple screenshots/screencasts of my UI changes
- [x] Client: I translated all the newly inserted strings into German and English

### Motivation and Context

https://github.com/ls1intum/Artemis/issues/1250

### Description
Apollon Diagrams now have a course assigned to them and only the diagrams that are part of the course will show up when looking for the diagrams in a course.

### Steps for Testing
1. Log in to Artemis
2. Navigate to Course Management
3. Navigate to Exercises
4. Click on Create Drag and Drop Model Quiz
5. Create diagrams in different courses and see if they are visible in courses other than the one they were created in. They shouldn't be visible.
-> old diagrams are not visible because there are so few on live that it isn't needed to put in an extra request for diagrams with no courseId 
6. Open a diagram and generate a quiz exercise from it.

### Screenshots

Bildschirmfoto von 2020-05-02 19-15-55

### CODE REVIEW

File: src/main/java/de/tum/in/www1/artemis/repository/ApollonDiagramRepository.java [hunk 1]
```diff
@@ -12,4 +15,6 @@
 @Repository
 public interface ApollonDiagramRepository extends JpaRepository<ApollonDiagram, Long> {
 
+    @Query("SELECT a FROM ApollonDiagram a WHERE a.courseId = :#{#courseId} OR a.courseId is null")
```
Comment (sascha11110) [inline src/main/java/de/tum/in/www1/artemis/repository/ApollonDiagramRepository.java:18]: I think the native query is not necessary here. `List<ApollonDiagram> findDiagramsByCourse(Long courseId);` should be enough and the JPA implementation will do all the magic.

E.g. compare to implementation in `TutorLeaderboardComplaintsViewRepository.java`.

File: src/test/java/de/tum/in/www1/artemis/ApollonDiagramResourceIntegrationTest.java [hunk 2]
```diff
@@ -90,6 +90,35 @@ public void testGetAllApollonDiagram_OK() throws Exception {
...
+        apollonDiagramRepository.save(newApollonDiagram);
+        List<ApollonDiagram> updatedResponse = request.get("/api/apollon-diagrams/list/1", HttpStatus.OK, List.class);
+        assertThat(updatedResponse.isEmpty()).as("updated response is not empty").isFalse();
+        assertThat(updatedResponse.size()).as("updated response has length 2").isEqualTo(2);
+
+        ApollonDiagram newerApollonDiagram = ModelFactory.generateApollonDiagram(DiagramType.ClassDiagram, "newer title");
+        newerApollonDiagram.setCourseId((long) 2);
+        apollonDiagramRepository.save(newerApollonDiagram);
+        updatedResponse = request.get("/api/apollon-diagrams/list/1", HttpStatus.OK, List.class);
+        assertThat(updatedResponse.isEmpty()).as("updateder response is not empty").isFalse();
```
Comment (sascha11110) [inline src/test/java/de/tum/in/www1/artemis/ApollonDiagramResourceIntegrationTest.java:113]: Typo

File: src/test/java/de/tum/in/www1/artemis/ApollonDiagramResourceIntegrationTest.java [hunk 3]
```diff
@@ -90,6 +90,35 @@ public void testGetAllApollonDiagram_OK() throws Exception {
...
+        assertThat(updatedResponse.size()).as("updateder response has length 2").isEqualTo(2);
```
Comment (sascha11110) [inline src/test/java/de/tum/in/www1/artemis/ApollonDiagramResourceIntegrationTest.java:114]: Typo

File: src/test/java/de/tum/in/www1/artemis/ApollonDiagramResourceIntegrationTest.java [hunk 4]
```diff
@@ -90,6 +90,35 @@ public void testGetAllApollonDiagram_OK() throws Exception {
         assertThat(updatedResponse.size()).as("updated response has length 2").isEqualTo(2);
     }
 
+    @Test
```
Comment (sascha11110) [inline src/test/java/de/tum/in/www1/artemis/ApollonDiagramResourceIntegrationTest.java:93]: In general you might test here that the evaluation of the courseId actually works. E.g. add two diagrams with different courses to the repository and test that only the one with the matching course is returned.

Comment (sjagla27) [reply src/test/java/de/tum/in/www1/artemis/ApollonDiagramResourceIntegrationTest.java:93]: As discussed.

**Comment by krusche** [comment]
If you click on Open, please make sure that you are still in the course context and that only instructors / tutors of the same can navigate into this (this includes all REST calls to the server, please use "isAtLeastTutorInCourse()").

Generate Quiz Exercise does not need the course selection any more

image

**Comment by krusche** [comment]
What is the status of this PR?

File: src/main/webapp/app/exercises/quiz/manage/apollon-diagrams/apollon-diagram-detail.component.ts [hunk 5]
```diff
@@ -9,6 +9,7 @@ import { convertRenderedSVGToPNG } from './exercise-generation/svg-renderer';
 import { ApollonDiagramService } from 'app/exercises/quiz/manage/apollon-diagrams/apollon-diagram.service';
 import { ApollonDiagram } from 'app/entities/apollon-diagram.model';
 import { AlertService } from 'app/core/alert/alert.service';
+import { courseExerciseOverviewTour } from 'app/guided-tour/tours/course-exercise-overview-tour';
```
Comment (TobiasPr) [inline src/main/webapp/app/exercises/quiz/manage/apollon-diagrams/apollon-diagram-detail.component.ts:12]: 'courseExerciseOverviewTour' is defined but never used - github action

**Pull request merged**